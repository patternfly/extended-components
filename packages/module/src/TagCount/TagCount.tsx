import React from 'react';
import clsx from 'clsx';
import { Button, ButtonProps, Icon } from '@patternfly/react-core';
import { TagIcon } from '@patternfly/react-icons';
import { createUseStyles } from 'react-jss'

const useStyles = createUseStyles({
  buttonTagCount: {
    color: 'var(--pf-t--color--gray--50)',
    display: 'flex',
    alignItems: 'center'
  },

  tagText: {
    marginLeft: 10
  }
});

export interface TagCountProps extends ButtonProps {
  /** Count to display in tag count component */
  count?: number;
  /** Additional classes added to the tag count component */
  className?: string;
  /** Icon size */
  iconSize?: 'sm' | 'md' | 'lg' | 'xl';
  /** Custom OUIA ID */
  ouiaId?: string | number;
}

const TagCount: React.FunctionComponent<TagCountProps> = ({
  count, 
  className,
  iconSize= 'md',
  ouiaId = 'TagCount',
  ...props 
}: TagCountProps) => {
  const classes = useStyles();
  const tagClasses = clsx(classes.buttonTagCount, className);
  return (
    <Button aria-label="Tag count" {...props} variant="plain" isDisabled={!count} className={tagClasses} ouiaId={ouiaId} {...props}>
      <Icon iconSize={iconSize} data-ouia-component-id={`${ouiaId}-icon`}>
        <TagIcon />
      </Icon>
      <span className={classes.tagText} data-ouia-component-id={`${ouiaId}-text`}>{count}</span>
    </Button>
  );
};

export default TagCount;
