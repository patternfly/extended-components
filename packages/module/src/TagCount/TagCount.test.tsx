import React from 'react';
import { render } from '@testing-library/react';
import TagCount from './TagCount';

describe('TagCount component', () => {
  it('should render a disabled tag count with no value', () => {
    expect(render(<TagCount />)).toMatchSnapshot();
  });


  it('should render a tag count of 11', () => {
    expect(render(<TagCount count={11} />)).toMatchSnapshot();
  });
});