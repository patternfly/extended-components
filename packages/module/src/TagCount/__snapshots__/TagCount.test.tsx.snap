// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCount component should render a disabled tag count with no value 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-disabled="false"
        aria-label="Tag count"
        class="pf-v6-c-button pf-m-plain buttonTagCount-0-2-1"
        data-ouia-component-id="TagCount"
        data-ouia-component-type="PF6/Button"
        data-ouia-safe="true"
        disabled=""
        type="button"
      >
        <span
          class="pf-v6-c-icon"
          data-ouia-component-id="TagCount-icon"
        >
          <span
            class="pf-v6-c-icon__content pf-m-md"
          >
            <svg
              aria-hidden="true"
              class="pf-v6-svg"
              fill="currentColor"
              height="1em"
              role="img"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
        </span>
        <span
          class="tagText-0-2-2"
          data-ouia-component-id="TagCount-text"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-disabled="false"
      aria-label="Tag count"
      class="pf-v6-c-button pf-m-plain buttonTagCount-0-2-1"
      data-ouia-component-id="TagCount"
      data-ouia-component-type="PF6/Button"
      data-ouia-safe="true"
      disabled=""
      type="button"
    >
      <span
        class="pf-v6-c-icon"
        data-ouia-component-id="TagCount-icon"
      >
        <span
          class="pf-v6-c-icon__content pf-m-md"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
            />
          </svg>
        </span>
      </span>
      <span
        class="tagText-0-2-2"
        data-ouia-component-id="TagCount-text"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TagCount component should render a tag count of 11 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-disabled="false"
        aria-label="Tag count"
        class="pf-v6-c-button pf-m-plain buttonTagCount-0-2-1"
        data-ouia-component-id="TagCount"
        data-ouia-component-type="PF6/Button"
        data-ouia-safe="true"
        type="button"
      >
        <span
          class="pf-v6-c-icon"
          data-ouia-component-id="TagCount-icon"
        >
          <span
            class="pf-v6-c-icon__content pf-m-md"
          >
            <svg
              aria-hidden="true"
              class="pf-v6-svg"
              fill="currentColor"
              height="1em"
              role="img"
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
              />
            </svg>
          </span>
        </span>
        <span
          class="tagText-0-2-2"
          data-ouia-component-id="TagCount-text"
        >
          11
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-disabled="false"
      aria-label="Tag count"
      class="pf-v6-c-button pf-m-plain buttonTagCount-0-2-1"
      data-ouia-component-id="TagCount"
      data-ouia-component-type="PF6/Button"
      data-ouia-safe="true"
      type="button"
    >
      <span
        class="pf-v6-c-icon"
        data-ouia-component-id="TagCount-icon"
      >
        <span
          class="pf-v6-c-icon__content pf-m-md"
        >
          <svg
            aria-hidden="true"
            class="pf-v6-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z"
            />
          </svg>
        </span>
      </span>
      <span
        class="tagText-0-2-2"
        data-ouia-component-id="TagCount-text"
      >
        11
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
